daffodil-core/src/main/scala/org/apache/daffodil/dsom/SimpleTypes.scala

```diff
@@ -40,14 +69,71 @@ trait NonPrimTypeMixin {
   }
 }
 
-sealed trait SimpleTypeBase extends TypeBase {
-  def primType: NodeInfo.PrimType = {
-    optRestriction.map { _.primType }.getOrElse {
-      optUnion.map { _.primType }.getOrElse {
-        Assert.invariantFailed("must be either a restriction or union")
+sealed trait SimpleTypeBase extends TypeBase
+  with hasOptRepTypeMixin {
+
+  def primType: PrimType
+
+}
+
+/*
+ * For components which can define dfdl:repValues and dfdl:repValueRanges
+ * Construct the repValueSet using only the above mentioned attributes on the element itself
+ * Applies to simpleType and enumeration
+ *
+ * In the case of simpleType, it is possible that optRepValueSetFromAttribute will be none
+ * but the element will still have an optRepValueSet for another source (eg. children elements)
+ */
+sealed trait HasRepValueAttributes extends AnnotatedSchemaComponent {
+  def optRepTypeFactory: Option[SimpleTypeFactory]
+  def optRepValueSet: Option[RepValueSet[AnyRef]]
+
+  lazy val (repValuesAttrCooked: Seq[AnyRef], repValueRangesAttrCooked: Seq[(RangeBound[BigInt], RangeBound[BigInt])]) = optRepTypeFactory match {
+    case Some(repType) => {
+      val repValueSetRaw = findPropertyOptionThisComponentOnly("repValues").toOption
+        .map(_.split("\\s+").toSeq).getOrElse(Seq())
+      val repValueRangesRaw = findPropertyOptionThisComponentOnly("repValueRanges").toOption
+        .map(_.split("\\s+").toSeq).getOrElse(Seq())
+      repType.primType match {
+        case PrimType.String => {
+          if (repValueRangesRaw.size > 0) context.SDE("repValueRanges set when using a string repType")
+          val repValueSetCooked = repValueRangesRaw.map(RepValueCooker.convertConstant(_, context, false))
+          (repValueSetCooked, Seq())
+        }
+        //I need to double check the type hierarchy, but I think all integer types are subtypes of SignedInteger
```

Comment:
That appears to be correct, so I would go ahead and remove this comment.

Reply:
There's a conversation regarding this in the dev mailing list. The TLDR is that SignedInteger isn't really being used, shouldn't exist, and has an inconsistent place in the XSD type hierarchy. 
But, I agree that this comment is correct from the perspective of the scala type system, so I'll remove it.